source/RoslynAnalyzers/Testing/Integration/IntegrationTestForwardCancellationTokenToInvocationsAnalyzer.cs

```diff
@@ -0,0 +1,271 @@
+                    {
+                        return;
+                    }
+
+                    var (cancellationTokenType, genericTaskType, genericValueTaskType) = GetWellKnownTypes(context);
+                    if (cancellationTokenType == null)
+                    {
+                        // Only ct type matters, for the generic type it is ok to be null (edge case).
+                        return;
+                    }
+
+                    var invocation = (IInvocationOperation)context.Operation;
+                    var method = invocation.TargetMethod;
+
+                    // Verify that the current invocation is not passing an explicit token already
+                    if (IsCancellationTokenAlreadyForwarded(invocation.Arguments, cancellationTokenType))
+                    {
+                        return;
+                    }
+
+                    // Verify method has optional ct, warn user to be explicit rather than relying on default.
+                    if (InvocationMethodTakesAToken(method, invocation.Arguments, cancellationTokenType) ||
+
+                        // Verify method has ct overload as last parameter.
+                        InvocationMethodHasCancellationTokenOverload(method, cancellationTokenType, genericTaskType, genericValueTaskType))
+                    {
+                        SyntaxNode? nodeToDiagnose = GetInvocationMethodNameNode(context.Operation.Syntax) ?? context.Operation.Syntax;
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptors.Oct2008IntegrationTestForwardCancellationTokenToInvocations(), nodeToDiagnose.GetLocation()));
+                    }
+                },
+                OperationKind.Invocation
+            );
+        }
+
+        static bool IsOperationOfInterest(OperationAnalysisContext context)
+        {
+            return context.ContainingSymbol is IMethodSymbol containingMethod &&
+
+                // Only interested in IntegrationTestType classes.
+                containingMethod.ContainingType.IsAssignableTo(new OctopusTestingContext(context.Compilation).IntegrationTestType) &&
+
+                // No diagnostic on static method, this is handled by CA2016 and Octopus async-cancellation-token convention.
+                !containingMethod.IsStatic;
+        }
+
+        static (INamedTypeSymbol? cancellationTokenType, INamedTypeSymbol? genericTaskType, INamedTypeSymbol? genericValueTaskType) GetWellKnownTypes(OperationAnalysisContext context)
+        {
+            INamedTypeSymbol? cancellationTokenType = context.Compilation.GetTypeByMetadataName(Constants.Types.SystemThreadingCancellationToken);
+            INamedTypeSymbol? genericTaskType = context.Compilation.GetTypeByMetadataName(Constants.Types.SystemThreadingTasksTask1);
+            INamedTypeSymbol? genericValueTaskType = context.Compilation.GetTypeByMetadataName(Constants.Types.SystemThreadingTasksValueTask1);
```

Comment:
Per other instances, I suspect the compiler would be hinting to use the preferred `var` here.

Reply:
Updated. The method definition clearly display the expected return type.